src/types.rs

```diff
@@ -98,13 +98,15 @@ impl FeeRate {
 
     /// Calculate fee rate from `fee` and weight units (`wu`).
     pub fn from_wu(fee: u64, wu: usize) -> FeeRate {
-        Self::from_vb(fee, wu.vbytes())
+        let vbytes = wu as f32 / 4.0;
+        let rate = fee as f32 / vbytes;
+        Self::new_checked(rate)
     }
 
     /// Calculate fee rate from `fee` and `vbytes`.
-    pub fn from_vb(fee: u64, vbytes: usize) -> FeeRate {
-        let rate = fee as f32 / vbytes as f32;
-        Self::from_sat_per_vb(rate)
+    pub fn from_vb(fee: u64, vbytes: f32) -> FeeRate {
+        let rate = fee as f32 / vbytes;
+        Self::new_checked(rate)
```

Comment:
I think using `Self::from_sat_per_vb(rate)` instead of  `Self::new_checked(rate)` is more descriptive